sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -13,13 +17,489 @@ namespace Azure.Iot.Hub.Service
+                response.GetRawResponse().Headers.TryGetValue(ContinuationTokenHeader, out string continuationToken);
+
+                return Page.FromValues(response.Value, continuationToken, response.GetRawResponse());
+            }
+
+            async Task<Page<TwinData>> NextPageFunc(string nextLink, int? pageSizeHint)
+            {
+                var querySpecification = new QuerySpecification();
+                Response<IReadOnlyList<TwinData>> response = await _registryManagerClient.QueryIotHubAsync(querySpecification, nextLink, pageSizeHint?.ToString(), cancellationToken).ConfigureAwait(false);
+                response.GetRawResponse().Headers.TryGetValue(ContinuationTokenHeader, out string continuationToken);
+                return Page.FromValues(response.Value, continuationToken, response.GetRawResponse());
+            }
+
+            return PageableHelpers.CreateAsyncEnumerable(FirstPageFunc, NextPageFunc);
+        }
+
+        /// <summary>
+        /// List a set of device twins.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A pageable set of device twins.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Globalization", "CA1305:Specify IFormatProvider", Justification = "<Pending>")]
+        public virtual Pageable<TwinData> GetTwins(CancellationToken cancellationToken = default)
+        {
+            Page<TwinData> FirstPageFunc(int? pageSizeHint)
+            {
+                var querySpecification = new QuerySpecification
+                {
+                    Query = "select * from devices"
+                };
+
+                Response<IReadOnlyList<TwinData>> response = _registryManagerClient.QueryIotHub(querySpecification, null, pageSizeHint?.ToString(), cancellationToken);
+
+                response.GetRawResponse().Headers.TryGetValue(ContinuationTokenHeader, out string continuationToken);
+
+                return Page.FromValues(response.Value, continuationToken, response.GetRawResponse());
+            }
+
+            Page<TwinData> NextPageFunc(string nextLink, int? pageSizeHint)
+            {
+                var querySpecification = new QuerySpecification();
+                Response<IReadOnlyList<TwinData>> response = _registryManagerClient.QueryIotHub(querySpecification, nextLink, pageSizeHint?.ToString(), cancellationToken);
+                response.GetRawResponse().Headers.TryGetValue(ContinuationTokenHeader, out string continuationToken);
+                return Page.FromValues(response.Value, continuationToken, response.GetRawResponse());
+            }
+
+            return PageableHelpers.CreateEnumerable(FirstPageFunc, NextPageFunc);
+        }
+
+        /// <summary>
+        /// Get a device's twin.
```

Comment:
just `device twin` ? not sure ...

Reply:
I don't follow. This is the get call for a device's twin. What else is it getting?

Reply:
we have "Device Twin" in multiple places, the 's looks strange to me. it may just be me, you can ignore this :D